astropy/nddata/utils.py

```diff
@@ -152,6 +169,28 @@ def overlap_slices(large_array_shape, small_array_shape, position, mode="partial
     return slices_large, slices_small
 
 
+@overload
+def extract_array(
+    array_large: NDArray[DT],
+    shape: int | tuple[int, ...],
+    position: float | tuple[float, ...],
+    mode: Literal["partial", "trim", "strict"],
+    fill_value: Complex,
```

Comment:
I'm not familiar enough with `nddata` to know if complex numbers are relevant here. If they are then this is annotation is correct. If they are not then we should use `numbers.Real` instead of `numbers.Complex`.

Reply:
Yeah I used Complex just to be on the safe side and stay as general as possible but I'd like subpackage maintainers to weigh in here specifically.